File: CS/MainViewModel.cs
using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;

namespace DXSample {
    public class MainViewModel : ViewModelBase {
        public bool HasErrors {
            get { return GetValue<bool>(); }
            set { SetValue(value); }
        }
        public ObservableCollection<Task> TaskList { get; }
        public MainViewModel() {
            TaskList = new ObservableCollection<Task> {
                new Task() {
                    TaskName = "Complete Project A",
                    StartDate = new DateTime(2009, 7, 17),
                    EndDate = new DateTime(2009, 7, 10)
                },
                new Task() {
                    TaskName = "Test Website",
                    StartDate = new DateTime(2009, 7, 10),
                    EndDate = new DateTime(2009, 7, 12)
                },
                new Task() {
                    TaskName = string.Empty,
                    StartDate = new DateTime(2009, 7, 4),
                    EndDate = new DateTime(2009, 7, 6)
                }
            };
        }

        [Command]
        public void ValidateRow(RowValidationArgs args) {
            args.Result = GetValidationErrorInfo((Task)args.Item);
        }
        static ValidationErrorInfo GetValidationErrorInfo(Task task) {
            if (task.StartDate > task.EndDate)
                return new ValidationErrorInfo("Start Date must be less than End Date");
            if (string.IsNullOrEmpty(task.TaskName))
                return new ValidationErrorInfo("Enter a task name");
            return null;
        }

        [Command]
        public void InvalidRow(InvalidRowExceptionArgs args) {
            args.ExceptionMode = ExceptionMode.NoAction;
        }
    }
}

File: VB/MainViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf

Namespace DXSample

    Public Class MainViewModel
        Inherits ViewModelBase

        Public Property HasErrors As Boolean
            Get
                Return GetValue(Of Boolean)()
            End Get

            Set(ByVal value As Boolean)
                SetValue(value)
            End Set
        End Property

        Public ReadOnly Property TaskList As ObservableCollection(Of Task)

        Public Sub New()
            TaskList = New ObservableCollection(Of Task) From {New Task() With {.TaskName = "Complete Project A", .StartDate = New DateTime(2009, 7, 17), .EndDate = New DateTime(2009, 7, 10)}, New Task() With {.TaskName = "Test Website", .StartDate = New DateTime(2009, 7, 10), .EndDate = New DateTime(2009, 7, 12)}, New Task() With {.TaskName = String.Empty, .StartDate = New DateTime(2009, 7, 4), .EndDate = New DateTime(2009, 7, 6)}}
        End Sub

        <Command>
        Public Sub ValidateRow(ByVal args As RowValidationArgs)
            args.Result = GetValidationErrorInfo(CType(args.Item, Task))
        End Sub

        Private Shared Function GetValidationErrorInfo(ByVal task As Task) As ValidationErrorInfo
            If task.StartDate > task.EndDate Then Return New ValidationErrorInfo("Start Date must be less than End Date")
            If String.IsNullOrEmpty(task.TaskName) Then Return New ValidationErrorInfo("Enter a task name")
            Return Nothing
        End Function

        <Command>
        Public Sub InvalidRow(ByVal args As InvalidRowExceptionArgs)
            args.ExceptionMode = ExceptionMode.NoAction
        End Sub
    End Class
End Namespace
